### DISCUSSION TITLE
Agisoft Metashape to NERF conversion

### DISCUSSION THREAD

**Discussion by EnricoAhlers** [NVlabs/instant-ngp#438] [Show and tell]
I wrote a simple script for converting Agisoft Metashape projects to NERF files. For me this yielded better results than using COLMAP, especially when there are very few images available.

You can find it here:
https://github.com/EnricoAhlers/agi2nerf

**Comment by bilawalsidhu** [comment]
When I use it I get an error - " RuntimeWarning: invalid value encountered in true_divide in line 41 -- totp /= totw" - do you have a fix for this?

**Reply by EnricoAhlers** [reply]
Could you please upload your XML file? I can take a look at it.

**Reply by Liu97022** [reply]
After I changed the line w > 0.01 into w > 0.0001, there is no error. Same error also have occurred in colmap2nerf. Please [see](https://github.com/NVlabs/instant-ngp/issues/434#issuecomment-1216106844).

**Reply by EnricoAhlers** [reply]
Thanks, fixed it!

**Reply by piorkoo** [reply]
Hi, I have the same error " RuntimeWarning: invalid value encountered in true_divide in line 41 -- totp /= totw" and the changed value w < 0.0001 doesn't work... Can you look at the attached xml and try to figure out whats is wrong?
[psdk_cameras.zip](https://github.com/NVlabs/instant-ngp/files/11424594/psdk_cameras.zip)

**Comment by felixgeen** [comment]
Hey Enciro, thanks for making this awesome tool. I wanted to ask if you know a good way of fixing the size of the model being too big? I keep getting results like:

```
computing center of attention...
[-19.47418558 -21.44102482   8.48883234]
```

Which is way outside the bounds of the nerf world. When i try scaling down the object inside metashape it doesnt change this center of attention, it comes out exactly the same no matter what i do with the scene inside metashape.

Thanks again!

**Reply by felixgeen** [reply]
![image](https://user-images.githubusercontent.com/32222984/178841504-5c1ca56a-9ead-4fb3-ac93-d95e13238e1e.png)
here you can see my camera paths are quite far outside my aabb 16 cube

**Reply by EnricoAhlers** [reply]
Hi, you should be able to adjust your json manually. Take a look at the instant-ngp docs. There is a scale and offset option.

**Reply by felixgeen** [reply]
thank you! By adding:
	"scale": 0.33,
	"offset": [0.5, 0.5, 0.5],
into the transforms.json just above "frames" it is possible to scale down and move around the scene.

**Comment by aburakkoc** [comment]
(nerfstudio) C:\Users\USER\Desktop\NS\data\oldport\agi2nerf>python agi2nerf.py --xml oldporthouse.xml
Traceback (most recent call last):
  File "agi2nerf.py", line 115, in <module>
    k1 = float(calibration[4].text)
IndexError: child index out of range

I am getting this error when I try to run the command line, I assume it is related to the XML file coming from meta shape. I used 360 photos and changed the camera calibration to spherical in meta shape. Is it because I used 360 images, if yes, is there a way to train 360 captured images?

thank you

**Reply by felixgeen** [reply]
I would also like to know this

**Reply by mvsc1** [reply]
To me this issue only occurred when not all images could be aligned in Metashape, ie number of input images is different from aligned images/cameras or cameras .XML references a different amount of images than is contained in the folder. 400+ images here.

**Comment by felixgeen** [comment]
Hi @EnricoAhlers, @Tom94  recently updated the code base to support equirectangular projection and latlong projection inside NGP.

> Simply add the key "equirectangular": true to transforms.json to enable it. (Similarly, latlong images are supported via "latlong": true.)

However, there doesn’t currently exist a solution for aligning and orientating a dataset of equirectangular images to bring into NGP. since Metashape has this ability, do you think it would be possible to update this awesome script you’ve made to support converting equirectangular alignments from Metashape into JSON?

**Reply by vassefer** [reply]
Do you know if this question has been answered? 
I'd be interested in working with equirectangular pictures too.
